mantidproject/mantid
# Handle empty sample names in processed NeXus

This PR changes Mantid's processed NeXus format such that an explicit attribute flag will be set if sample name is an empty string. This is to overcome the issue that the NeXus library replaces an empty attribute string by a space character `' '`.

This change should not affect loading of previous NeXus files. Besides the new attribute, the saved NeXus files should be same as before.

**Report to:** [nobody].

**To test:**

Make sure the following script runs without issues after applying these changes. Please change the file path to point to something sensible on your system.

```python
ws = CreateSampleWorkspace()
SaveNexusProcessed(ws, '/users/soininen/delete_me.nxs')
loaded = LoadNexusProcessed('/users/soininen/delete_me.nxs')

sampleOriginal = ws.sample().getName()
sampleLoaded = loaded.sample().getName()
print('"{}"'.format(sampleOriginal))
print('"{}"'.format(sampleLoaded))
```

Inspect the contents of the Nexus file by `hdfview` or similar. Check that `/mantid_workspace_1/sample` has an attribute `name_empty` with a value of 1.

Fixes #24712.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
